### PR TITLE
@deck.gl/json: expression parser for json and pydeck

### PR DESCRIPTION
**Pull request by ibgreen**
# For #3079 

@ajduberstein Found a small expression parser that does handle array indexing, ternary experssions etc. See overrides of accessors in screenshot.

Screen Shot 2019-06-04 at 7 08 17 AM

For #

#### Change List
-

### CODE REVIEW

**Comment by Pessimistress** [comment]
I'd be very cautious to evaluate arbitrary user-provided expressions because this imposes a serious security risk.

- The json module documentation must call out the risk and caution users to only load layers and data from trusted source.
- In the Jupyter integration use case, someone can potentially use this attack the Jupyter notebook server. Do we have an evaluation of the sandboxing mechanism of Jupyter widgets?

**Comment by ibgreen** [comment]
Yeah the plan is to leverage the ability of the chosen library to return an AST and remove function call nodes. We will either disallow them completely or only support a handful of built in math functions (like `Math.abs`...), but no general function expressions.

The key expression parsing feature is array indexing and creation, not function calls.

**Comment by ibgreen** [comment]
I also am thinking that we should copy in the expression parser code into our repo. Too risky to have it this a dependency outside of our control. It is just one file, not very big.

File: modules/json/src/parsers/convert-json-expression.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  if (isSingleIdentifier) {
+    func = row => get(row, propValue);
+  } else {
+    // Compile with expression-eval
+    try {
+      const ast = expressionEval.parse(propValue);
+      // NOTE: To avoid security risks, the arguments passed to the
+      // compiled expresion must only give access to pure data (no globals etc)
+      // We disable function call syntax?
+      traverse(ast, node => {
```
Comment (ibgreen) [inline modules/json/src/parsers/convert-json-expression.js]: @ajduberstein Per our discussion today. This should catch any function calls and throw an exception, preventing execution. We'll need to add a couple of test cases to ensure this filter is working.

**Comment by ajduberstein** [comment]
Addressed by #3397

**Pull request closed without merging**